core/src/main/java/org/elasticsearch/script/ScriptSettings.java

```diff
@@ -160,15 +160,19 @@ public ScriptSettings(ScriptEngineRegistry scriptEngineRegistry, ScriptContextRe
         settings.addAll(SCRIPT_TYPE_SETTING_MAP.values());
         settings.addAll(scriptContextSettingMap.values());
         settings.addAll(scriptLanguageSettings);
-        settings.add(defaultScriptLanguageSetting);
+        settings.add(defaultLegacyScriptLanguageSetting);
         return settings;
     }
 
     public Iterable<Setting<Boolean>> getScriptLanguageSettings() {
         return scriptLanguageSettings;
     }
 
-    public Setting<String> getDefaultScriptLanguageSetting() {
-        return defaultScriptLanguageSetting;
+    public Setting<String> getDefaultLegacyScriptLanguageSetting() {
+        return defaultLegacyScriptLanguageSetting;
+    }
+
+    public static String getLegacyDefaultLang(Settings settings) {
+        return settings.get("script.legacy.default_lang", ScriptSettings.LEGACY_DEFAULT_LANG);
```

Comment:
this key is different from the one specified above in https://github.com/elastic/elasticsearch/pull/20310/files#diff-9a1a4781a78e7e7a89657e8717f88997R61